codes/cpp/include/PrintUtil.hpp

```diff
@@ -293,4 +293,48 @@ class PrintUtil {
                 cout << kv.first << " -> " << kv.second << '\n';
             }
         }
+
+        /**
+         * @brief Print a Max-Heap (PriorityQueue)
+         *
+         * @tparam T
+         * @param heap
+         */
+        template <typename T>
+        static void printHeap(priority_queue<T, vector<T>, less<T>> heap)
+        {
+            vector<int> vec;
+            while (!heap.empty())
+            {
+                vec.push_back(heap.top());
+                heap.pop();
+            }
+            cout << "堆的数组表示：" << endl;
+            printVector(vec);
+            cout << "堆的树状表示：" << endl;
+            TreeNode *root = vecToTree(vec);
+            printTree(root);
+        }
+
+        /**
+         * @brief Print a Min-Heap (PriorityQueue)
+         *
+         * @tparam T
+         * @param heap
+         */
+        template <typename T>
+        static void printHeap(priority_queue<T, vector<T>, greater<T>> heap)
+        {
+            vector<int> vec;
+            while (!heap.empty())
+            {
+                vec.push_back(heap.top());
+                heap.pop();
```

Comment:
It is not correct because the `vec` will become an ordered list, which should be an original heap.

Reply:
I haven't thought of a good way to print out the values corresponding to the priority_queue as an array here, is there a relevant function in C++ to convert the priority_queue to an array? (like the ArrayList provided by Java), can you provide some ideas?

Reply:
https://www.appsloveworld.com/cplus/100/1503/converting-a-priority-queue-to-vector-of-ints-in-c

Could you try to derive from priority_queue and expose the `std:vector` container?

Reply:
I've revised it (sorry for taking so long to update because I couldn't help myself in Chinese New Year)
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/52858395/216065555-d7406b53-12d4-4044-b620-257db6b99453.png">

Reply:
The rest will be done in the next few days

Reply:
> The rest will be done in the next few days

Great!